### PR TITLE
Mobileapps 1238

### PR DESCRIPTION
**Pull request by aman-alfresco**
- add pageView events

### CODE REVIEW

File: data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt [hunk 1]
```diff
@@ -111,11 +110,40 @@ enum class EventName(val value: String) {
...
+    ShareExtension("PageView_ShareExtension"),
+    Transfers("PageView_Transfers"),
+    None("none")
+}
+
+/**
+ * Marked as APIEvent enum
+ */
+enum class APIEvent(val value: String) {
+    SyncingStatus("Event_API_SyncingStatus"),
```
Comment (ankitgoyalgloballogic) [inline data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt:136]: As discussed, no need of this sync event

File: data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt [hunk 2]
```diff
@@ -111,11 +110,40 @@ enum class EventName(val value: String) {
     OpenWith("Event_openWith")
 }
 
+/**
+ * Marked as PageView enum
+ */
+enum class PageView(val value: String) {
+    Recent("PageView_Recent"),
```
Comment (ankitgoyalgloballogic) [inline data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt:117]: If possible, can we add the prefix "PageView_" in the function only?

File: data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt [hunk 3]
```diff
@@ -111,11 +110,40 @@ enum class EventName(val value: String) {
...
+    Transfers("PageView_Transfers"),
+    None("none")
+}
+
+/**
+ * Marked as APIEvent enum
+ */
+enum class APIEvent(val value: String) {
+    SyncingStatus("Event_API_SyncingStatus"),
+    NewFolder("Event_API_NewFolder"),
```
Comment (ankitgoyalgloballogic) [inline data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt:137]: If possible, can we add the prefix "Event_API_" in the function only?

**Review by ankitgoyalgloballogic — CHANGES_REQUESTED** [review]

File: data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt (hunk 2 above)
Comment (aman-alfresco) [reply data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt:117]: It's possible but that would be a string type of value that we will pass to function but right now, I am accepting it as predefined enum

File: data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt (hunk 3 above)
Comment (aman-alfresco) [reply data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt:137]: It's possible but that would be a string type of value that we will pass to function but right now, I am accepting it as predefined enum

File: data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt (hunk 1 above)
Comment (aman-alfresco) [reply data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt:136]: I'll remove the same in the next ticket(MOBILEAPPS-1239)

File: data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt (hunk 2 above)
Comment (ankitgoyalgloballogic) [reply data/src/main/kotlin/com/alfresco/content/data/AnalyticsRepository.kt:117]: Use "Recent" as the predefined enum and pass it in the function. Add prefix in the called function only. Thoughts?

**Review by gl-lovekesh — APPROVED** [review]

**Pull request merged**